tests/cbtest/gtest/test_uzfs.cc

```diff
@@ -410,6 +410,129 @@ uzfs_mock_rebuild_scanner_rebuild_comp(void *arg)
 	zk_thread_exit();
 }
 
+void
+verify_rebuild_io(int rebuild_fd, zvol_info_t *zinfo)
+{
+	int rc;
+	char *write_buf, *read_buf, *send_buf;
+	char *cbuf;
+	int clen;
+	int len = 4096;
+	metadata_desc_t *md;
+	time_t now;
+	uint64_t cnt;
+	zvol_io_hdr_t rebuild_hdr = { 0 };
+	struct zvol_io_rw_hdr *rw_hdr;
+
+	now = time(0);
+	cbuf = ctime(&now);
+	clen = strlen(cbuf) - 1;
+
+	rebuild_hdr.opcode = ZVOL_OPCODE_READ;
+	rebuild_hdr.flags = ZVOL_OP_FLAG_REBUILD;
+	rebuild_hdr.status = ZVOL_OP_STATUS_OK;
+	rebuild_hdr.offset = 4096;
+	rebuild_hdr.len = len + sizeof (struct zvol_io_rw_hdr);
+
+	rc = uzfs_zvol_socket_write(rebuild_fd, (char *)&rebuild_hdr, sizeof(rebuild_hdr));
+        EXPECT_NE(rc, -1);
+
+	send_buf = (char *)malloc(rebuild_hdr.len);
+	write_buf = send_buf + sizeof (struct zvol_io_rw_hdr);
+
+	for (int i = 0; i < len; i += clen) {
+		memcpy(write_buf + i, cbuf, clen);
```

Comment:
mem overwrite can happen here.. say i is less than len, but, i+clen can be more than len

Reply:
ok